tools/flight_recorder/fr_trace.py

```diff
@@ -170,6 +187,50 @@ class Database(NamedTuple):
 }
 
 
+class MatchState(Enum):
+    """
+    Enum representing the possible states of matching for collective operations.
+
+    - FULLY_MATCHED: Indicates that all aspects of the collective operations match.
```

Comment:
Nit: these strings seem useful for reporting error type to the user too.  should we attach them to the enum somehow or make a dict from enum to string to be able to print these out to the user at the appropriate time?

Reply:
I see. Yeah let's add it.